cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/messaging/TopicMessageIdStore.java

```diff
@@ -25,21 +25,22 @@
  */
 public interface TopicMessageIdStore {
   /**
-   * Gets the id of the last fetched message that was set for a given TMS topic. This method is called before
-   * processing messages from TMS so that the subscriber fetches previously unseen notifications.
+   * Gets the id of the last fetched message that was set for a subscriber of the given TMS topic
    *
    * @param topic the topic to lookup the last message id
-   * @return the id of the last fetched message for this topic, or null if no message id was stored from the given topic
+   * @param subscriber the subscriber name
+   * @return the id of the last fetched message for this subscriber on this topic,
+   *         or {@code null} if no message id was stored before
    */
   @Nullable
-  String retrieveSubscriberState(String topic);
```

Comment:
can keep the original method as a shorthand for empty subscriber name?

Reply:
No, why?

Reply:
We actually don't need this interface, do you know why it was introduced?

Reply:
it's just for getting the last message ID which the subscriber has processed so far. It's only implemented by `JobQueueDataset` and `AppMetadataStore`. You are right, the interface is not really necessary.

Reply:
I removed it.